test/format/snap.c

```diff
@@ -405,6 +405,7 @@ snap_verify(TINFO *tinfo, SNAP_OPS *snap)
     CUR2S(cursor)->format_private = NULL;
 
     testutil_assert(ret == 0 || ret == WT_NOTFOUND || ret == WT_ROLLBACK);
+    testutil_check(cursor->reset(cursor));
```

Comment:
I'd be very hesitant to do this.  If this were application code, I'd agree to this -- keep the cursor clean, release resources between uses.  But this is test code, and we want to cover the bases where developers do not always clean up the cursor (or can't do so).  Doing a reset unsets the cursor position, so it may release a pin on a memory page.  That may give the program more freedom in allowing eviction, etc. I think that would give less code coverage. It also may release some memory for key/value (perhaps a minor effect).

Why do we need/want this call, to unset bounds?  I wouldn't expect bounds to be set, it looks like bounds are cleared at the end of places where they are used.

Reply:
Sorry @ddanderson let me give you some context to explain why I placed a cursor reset here. Initially as part of applying bounds to the `table_ops` function, I was using the `positioned` variable to make sure that we only apply bounds to an unpositioned cursor. Through adding the condition I was finding errors that the cursor was still positioned when trying to apply bounds on an "unpositioned" marked cursor by test/format. 

Through investigating this position, it let me to believe that the "positioned" variable doesn't accurately reflect if the cursor is positioned or not. I find that this is because `snap_verify` was taking a cursor and performing a search to check for the snapshot isolation mismatch but never frees the position of the cursor. This becomes a bigger problem when `table_op` assumes that any cursor we grab will not be positioned. Therefore I added an reset() here since it reflects an actual variable correctness issue rather than something to be specific to bounds.

Reply:
I'd like to find a resolution that leaves the `reset` out of this code, because of the code coverage issue.  (unless you can convince me that this is not important).  If the `positioned` variable is incorrect, perhaps fix where that is being tracked wrong?  Or perhaps call `reset` before applying the bounds?  I think either of these solutions address the code coverage issue.

Reply:
Personally I understand the benefit of leaving the cursor pinned. I will create a follow-up ticket to better track the `positioned` variable, since it seems to be a problem that is more test/format rather than the project.